tuf/data/roles_test.go

```diff
@@ -105,44 +105,61 @@ func TestErrInvalidRole(t *testing.T) {
-		path.Join("  ", CanonicalTargetsRole, "level1")))
-
-	require.False(t, IsDelegation(
-		path.Join("  "+CanonicalTargetsRole, "level1")))
-
-	require.False(t, IsDelegation(
-		path.Join(CanonicalTargetsRole, "level1"+"  ")))
+	f := require.False
+	tr := require.True
+	for val, check := range map[string]func(require.TestingT, bool, ...interface{}){
+		// false checks
+		path.Join(CanonicalTargetsRole, strings.Repeat("x", 255-len(CanonicalTargetsRole))): f,
+		"":                                                                                  f,
+		CanonicalRootRole:                                                                   f,
+		path.Join(CanonicalRootRole, "level1"):                                              f,
+		CanonicalTargetsRole:                                                                f,
+		CanonicalTargetsRole + "/":                                                          f,
+		path.Join(CanonicalTargetsRole, "level1") + "/":                                     f,
+		path.Join(CanonicalTargetsRole, "UpperCase"):                                        f,
+		path.Join(CanonicalTargetsRole, "directory") + "/../../traversal":                   f,
+		CanonicalTargetsRole + "///test/middle/slashes":                                     f,
+		CanonicalTargetsRole + "/./././":                                                    f,
+		path.Join("  ", CanonicalTargetsRole, "level1"):                                     f,
+		path.Join("  "+CanonicalTargetsRole, "level1"):                                      f,
+		path.Join(CanonicalTargetsRole, "level1"+"  "):                                      f,
+		path.Join(CanonicalTargetsRole, "white   space"+"level2"):                           f,
+		path.Join(CanonicalTargetsRole, strings.Repeat("x", 256-len(CanonicalTargetsRole))): f,
+
+		// true checks
+		path.Join(CanonicalTargetsRole, "level1"):                     tr,
+		path.Join(CanonicalTargetsRole, "level1", "level2", "level3"): tr,
+		path.Join(CanonicalTargetsRole, "under_score"):                tr,
+		path.Join(CanonicalTargetsRole, "hyphen-hyphen"):              tr,
+	} {
+		check(t, IsDelegation(val))
+	}
 
-	require.False(t, IsDelegation(
-		path.Join(CanonicalTargetsRole, "white   space"+"level2")))
+}
 
-	require.False(t, IsDelegation(
-		path.Join(CanonicalTargetsRole, strings.Repeat("x", 256-len(CanonicalTargetsRole)))))
+func TestIsWildDelegation(t *testing.T) {
+	f := require.False
+	tr := require.True
+	for val, check := range map[string]func(require.TestingT, bool, ...interface{}){
+		// false checks
+		CanonicalRootRole:      f,
+		CanonicalTargetsRole:   f,
+		CanonicalSnapshotRole:  f,
+		CanonicalTimestampRole: f,
+		"foo":   f,
+		"foo/*": f,
+		path.Join(CanonicalRootRole, "*"):           f,
+		path.Join(CanonicalSnapshotRole, "*"):       f,
+		path.Join(CanonicalTimestampRole, "*"):      f,
+		path.Join(CanonicalTargetsRole, "*", "foo"): f,
+		path.Join(CanonicalTargetsRole, "*", "*"):   f,
```

Comment:
Can also add failure cases for trailing slashes after the `*`, and extra slashes between the rest of the delegation and the `*`?